python/looker_sdk/rtl/model.py

```diff
@@ -24,7 +24,7 @@
 """
 
 import collections
-from typing import Any, cast, Iterable, MutableSequence, Optional, TypeVar
+from typing import Any, cast, Iterable, Sequence, Optional, TypeVar
```

Comment:
Is this to make those collections read-only? Seems like a good approach.

Reply:
Unfortunately this is complicated... here's a verbose explanation (mostly for my own benefit):

I originally thought I needed `MutableSequence` because it inherits `Sequence` methods and adds `append`, `reverse`, `extend`, `pop`, `remove` and I'd wanted to be flexible to the end user as to how they constructed it. But most of the API models that have list-like properties are typed Sequence so you can't just do

```python
sdk.all_groups(ids=models.DelimSequence(user.group_ids))
```

because `user.group_ids` is of type `Sequence` which is a super type of `MutableSequence` so you get

```
Argument 1 to "DelimSequence" has incompatible type "Sequence[int]"; expected "MutableSequence[int]
``` 

But now I realize that `collections.UserList` gives me those mutation methods methods (unfortunately I have to override them to get typing enforcement) and it's ok to just use `Sequence[T]`